recipe-box-backend/src/main/java/com/endava/recipebox/controller/RecipeController.java

```diff
@@ -1,51 +1,36 @@
 package com.endava.recipebox.controller;
 
-import com.endava.recipebox.model.MealType;
-import com.endava.recipebox.response.CustomRecipeMapper;
+
+import com.endava.recipebox.response.CustomRecipe;
 import com.endava.recipebox.service.RecipeService;
-import org.mapstruct.factory.Mappers;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.PathVariable;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.*;
 
-import java.io.IOException;
 
+import java.util.List;
 
 
 @RestController
-@RequestMapping(path = "/recipe_box", produces = "application/json")
+@RequestMapping(path = "/recipes", produces = "application/json")
 public class RecipeController {
 
-    public final RecipeService recipeService;
-    private final CustomRecipeMapper customRecipeMapperMapper = Mappers.getMapper(CustomRecipeMapper.class);
+    private final RecipeService recipeService;
 
     @Autowired
     RecipeController(RecipeService recipeService) {
         this.recipeService = recipeService;
     }
 
-    @GetMapping("/publicRecipes")
-    public ResponseEntity<?> getAllPublicRecipes() throws IOException {
-        return ResponseEntity.ok(customRecipeMapperMapper.map(recipeService.getAllPublicRecipes()));
+    @GetMapping("/")
+    public ResponseEntity<List<CustomRecipe>> getAllPublicRecipes(@RequestParam(required = false) Long id) {
```

Comment:
The requested param should be mealType. Because we need to filter all public recipes by mealType.